File: src/main/java/example/domain/model/rules/AdditionalSurchargeTable.java
package example.domain.model.rules;

import example.domain.model.specification.Destination;
import org.springframework.stereotype.Component;

import java.util.EnumMap;
import java.util.Map;

/**
 * のぞみ割増特急券テーブル
 */
@Component
public class AdditionalSurchargeTable {

    Map<Destination, Integer> map =new EnumMap<Destination, Integer>(Destination .class);

    {
        map.put(Destination.新大阪, 320);
        map.put(Destination.姫路, 530);
    }

    public int surcharge(Destination destination) {
        return map.get(destination);
    }
}

File: jr-pricing/Model/Rules/AdditionalSurchargeTable.vb
Imports Jr.Pricing.Model.Specification

Namespace Model.Rules
    ''' <summary>
    ''' のぞみ割り増し特急券テーブル
    ''' </summary>
    ''' <remarks></remarks>
    Public Class AdditionalSurchargeTable

        Private ReadOnly map As New Dictionary(Of Destination, Integer) From {
            {Destination.新大阪, 320},
            {Destination.姫路, 530}
        }

        Public Function GetAdditionalSurcharge(destination As Destination) As Integer
            Return map(destination)
        End Function

    End Class
End Namespace
